Support IP addresses with subnets in allowed_hosts

mleiner opened the pull request:
This fixes Issue #46.

allowed_hosts of nrpe.cfg can be set to
- hostname
- ip addresses (v4 and v6)
- ip addresses with subnet (v4 and v6)

The current code uses Stdlib::Host which ultimatively resolves to FQDNs and pure IP addresses, but not subnets. It uses Stdlib::Compat, too.

Changing Stdlib::Host to Variant[Stdlib::Fqdn,Stdlib::IP::Address] solves this by
- Resolve Stdlib::Host to Stdlib::Fqdn,Stdlib::Compat::Ip_address
- In the same step replace Stdlib::Compat::Ip_address to Stdlib::IP::Address

This gets rid of the Stdlib::Compat layer and allows `allowed_hosts` to accept IP addresses with subnets, too.

I have not found any spec tests for allowed_hosts, thus didn't change existing ones. I can try to add those if you like.

I've tested locally with 
```
nrpe::allowed_hosts:
  - "::1"
  - "2001::/96"
  - "127.0.0.1"
  - "192.168.0.0/24"
```

Improvement suggestions welcome

egmont1227 commented:
@alexjfisher @bastelfreak @blkperl can you have a look and merge this ?

bastelfreak commented on manifests/init.pp:
```diff
@@ -100,7 +100,7 @@
 #   **Private** You should not need to override this parameter.
 class nrpe (
-  Array[Stdlib::Host]                  $allowed_hosts                   = ['127.0.0.1'],
+  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]]                  $allowed_hosts                   = ['127.0.0.1'],
```
```suggestion
  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]]  $allowed_hosts                   = ['127.0.0.1'],
```

can you please clean it up a bit like this this and maybe remove a bit of the whitespace so the `=` are vertically aligned?

mleiner replied:
This was unintended, this is now fixed.

bastelfreak commented on templates/nrpe.cfg.epp:
```diff
@@ -5,7 +5,7 @@
   String                         $nrpe_user,
   String                         $nrpe_group,
-  Array[Stdlib::Host]            $allowed_hosts,
+  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]]            $allowed_hosts,
```
```suggestion
  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]] $allowed_hosts,
```

bastelfreak commented:
thanks for the work! can you please add some tests in https://github.com/voxpupuli/puppet-nrpe/blob/master/spec/classes/nrpe_spec.rb that verify it works with the different ip/cidr/fqdn formats?

mleiner commented:
@bastelfreak I've fixed the syntax and added positive spec tests for allowed_hosts. I skipped negative tests since I expect stdlib to catch type mismatches. Is that correct?

bastelfreak commented:
looks good to me. can you take a look at the indentation in templates/nrpe.cfg.epp ?

mleiner commented:
I have fixed the template syntax, too. Sorry for not checking while I already worked on init.pp.

bastelfreak commented:
thanks!

mleiner commented:
Are you planning a new release?

bastelfreak commented:
I would like to, but that's currently blocked by https://github.com/voxpupuli/puppet-nrpe/pull/48
